lib/elixir/lib/inspect/algebra.ex

```diff
@@ -283,6 +299,23 @@ defmodule Inspect.Algebra do
     fold_doc(docs, &concat(&1, &2))
   end
 
+  @doc """
+  Colors a document if a the identifier has a color in the options.
+  """
+  @spec color(color_identifier, t, Inspect.Opts.t) :: doc_color
+  def color(identifier, doc, opts = %Inspect.Opts{}) when is_doc(doc) do
+    precolor = Keyword.get(opts.syntax_colors, identifier)
```

Comment:
You can match `%Inspect.Opts{syntax_colors: syntax_colors}` at the signature since we will use it anyway. :D